backend-api/template.yaml

```diff
@@ -290,6 +290,53 @@ Resources:
       LogGroupName: !Sub /aws/apigateway/${AWS::StackName}-private-api-access-logs
       RetentionInDays: 30
 
+  PrivateApiDomainName:
+    Type: AWS::ApiGateway::DomainNameV2
+    Properties:
+      DomainName: !Sub
+        - private-${AWS::StackName}.${DNS_RECORD}
+        - DNS_RECORD: !FindInMap
+            - DNS
+            - !Ref Environment
+            - BaseDns
+      EndpointConfiguration:
+        Types:
+          - PRIVATE
+      CertificateArn: !Sub '{{resolve:ssm:/${Environment}/Platform/ACM/AsyncPrimaryZoneWildcardCertificateARN}}'
+      SecurityPolicy: TLS_1_2
+    
+  PrivateApiBasePathMapping:
+    Type: AWS::ApiGateway::BasePathMappingV2
+    DependsOn: PrivateApiDomainName
+    Properties:
+      DomainNameArn: !GetAtt PrivateApiDomainName.DomainNameArn
+      RestApiId: !Ref PrivateApi
+      Stage: !Ref PrivateApi.Stage
+
+  PrivateApiDomainNameAccessAssociation: 
+    Type: AWS::ApiGateway::DomainNameAccessAssociation
+    DependsOn: PrivateApiDomainName
+    Properties:
+      DomainNameArn: !GetAtt PrivateApiDomainName.DomainNameArn
+      AccessAssociationSource: vpce-0825f978898f57283
+      AccessAssociationSourceType: VPCE
+    
+  PrivateApiRecordSet:
+    Type: AWS::Route53::RecordSet
+    DependsOn: PrivateApiDomainName
+    Properties:
+      Name: !Sub
+        - private-${AWS::StackName}.${DNS_RECORD}
+        - DNS_RECORD: !FindInMap
+            - DNS
+            - !Ref Environment
+            - BaseDns
+      Type: A
+      HostedZoneId: !ImportValue PublicHostedZoneId #Private hosted zone needed here
+      AliasTarget:
+        DNSName: !Sub ${PrivateApi}.execute-api.${AWS::Region}.amazonaws.com/${Environment}
+        HostedZoneId: !ImportValue PublicHostedZoneId #Private hosted zone needed here
```

Comment:
336 and 338 need values for a private hosted zone, we currently only have a public hosted zone in our dns template.